private/bufpkg/bufconnect/interceptors.go

```diff
@@ -43,64 +44,123 @@ func NewSetCLIVersionInterceptor(version string) connect.UnaryInterceptorFunc {
-// user is already authenticated and the token is stored in .netrc)
+// which will set the auth token into the request header by the provided option.
 //
 // Note that the interceptor returned from this provider is always applied LAST in the series of interceptors added to
 // a client.
-func NewAuthorizationInterceptorProvider(container app.EnvContainer) func(string) connect.UnaryInterceptorFunc {
+func NewAuthorizationInterceptorProvider(option AuthorizeOption) func(string) connect.UnaryInterceptorFunc {
 	return func(address string) connect.UnaryInterceptorFunc {
 		interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
 			return connect.UnaryFunc(func(
 				ctx context.Context,
 				req connect.AnyRequest,
 			) (connect.AnyResponse, error) {
-				envKey := tokenEnvKey
-				token := container.Env(envKey)
-				if token == "" {
-					envKey = ""
-					machine, err := netrc.GetMachineForName(container, address)
-					if err != nil {
-						return nil, fmt.Errorf("failed to read server password from netrc: %w", err)
-					}
-					if machine != nil {
-						token = machine.Password()
-					}
-				}
-				if token != "" {
-					req.Header().Set(AuthenticationHeader, AuthenticationTokenPrefix+token)
-				}
-				response, err := next(ctx, req)
-				if err != nil {
-					err = &ErrAuth{cause: err, tokenEnvKey: envKey}
-				}
-				return response, err
+				return option(req, ctx, next, address)
 			})
 		}
 		return interceptor
 	}
 }
 
-// NewAuthorizationInterceptorProviderWithToken returns a new provider function which, when invoked, returns an
-// interceptor which sets the provided auth token into the request header.  This is used for registry providers where
-// the token is known at provider creation (i.e. when logging in and explicitly pasting a token into stdin
-//
-// Note that the interceptor returned from this provider is always applied LAST in the series of interceptors added to
-// a client.
-func NewAuthorizationInterceptorProviderWithToken(token string) func(string) connect.UnaryInterceptorFunc {
-	return func(address string) connect.UnaryInterceptorFunc {
-		interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
-			return connect.UnaryFunc(func(
-				ctx context.Context,
-				req connect.AnyRequest,
-			) (connect.AnyResponse, error) {
-				if token != "" {
-					req.Header().Set(AuthenticationHeader, AuthenticationTokenPrefix+token)
-				}
-				return next(ctx, req)
-			})
+// AuthorizeOption is an option for NewAuthorizationInterceptorProvider.
+type AuthorizeOption func(connect.AnyRequest, context.Context, connect.UnaryFunc, string) (connect.AnyResponse, error)
+
+// AuthorizeWithProvidedToken returns a new SetAuthTokenOption that will set the
+// provided token into the request header This is used for registry providers where the token is known at provider
+// creation (i.e. when logging in and explicitly pasting a token into stdin
+func AuthorizeWithProvidedToken(token string) AuthorizeOption {
```

Comment:
Doc string should have a consistent column width where newlines start, for example the newlines should start at the same column as the end of `... that will set the`

Also the doc string should be consistent about closing `)` and terminating with a period.